### PR TITLE
Change skill logic

### PR DESCRIPTION
**Pull request by TehDmitry**
* use meteor by cooldown in center-line
* use blackhole in center-line
* use bomb if enemy cnt > 3
* select more effective target for bomb
* reduce enemy scoring if they will be affected by meteor

### CODE REVIEW

File: index.user.js [hunk 1]
```diff
@@ -234,6 +237,43 @@ const EnemyWillAffectedByBoulder = function EnemyWillAffectedByBoulder(enemy) {
...
+    return Math.sqrt( Math.pow((x1-x2), 2) + Math.pow((y1-y2), 2) );
+}
+const DistBetweenSpriteCenters = function DistBetweenSpriteCenters(sprite1, sprite2) {
+    return DistBetweenPoints(SpriteCenter(sprite1)[0], SpriteCenter(sprite1)[1], SpriteCenter(sprite2)[0], SpriteCenter(sprite2)[1]);
+}
+const EnemyIsAffectedByBlackhole = function EnemyIsAffectedByBlackhole(enemy) {
+    if(GAME.m_State.m_AttackManager.m_mapBlackholes.size > 0) {
+        let hole = GAME.m_State.m_AttackManager.m_mapBlackholes.values().next().value;
+        let dist = DistBetweenSpriteCenters(hole, enemy.m_Sprite);
+        if(dist > 200 && dist < 250) {
```
Comment (EoD) [inline index.user.js]: I don't think this is the way the game currently checks if a enemy is stuck in a blackhole. See the following link on how it is done ingame:
https://steamcommunity-a.akamaihd.net/public/javascript/saliengame/attacks.js
```js
var rgEnemies = instance.m_BattleState.GetEnemiesAtLocation( new PIXI.Rectangle( blackhole.x - (blackhole.width / 2), blackhole.y - ( blackhole.height / 2 ), blackhole.width, blackhole.height ) );
rgEnemies.forEach( function( enemy ){
	if ( ( enemy.m_Sprite.x + enemy.m_Sprite.width / 2 ) < targetx )
	{
		enemy.m_Sprite.x += instance.m_AttackData.blackhole.strength;
	}
	if ( ( enemy.m_Sprite.x + enemy.m_Sprite.width / 2 ) > targetx )
	{
		enemy.m_Sprite.x -= instance.m_AttackData.blackhole.strength;
	}
	if ( ( enemy.m_Sprite.y + enemy.m_Sprite.height / 2 ) < targety )
	{
		enemy.m_Sprite.y += instance.m_AttackData.blackhole.strength;
	}
	if ( ( enemy.m_Sprite.y + enemy.m_Sprite.height / 2 ) > targety )
	{
		enemy.m_Sprite.y -= instance.m_AttackData.blackhole.strength;
	}
});
```

I made something similar in https://github.com/meepen/salien-bot/pull/76/commits/1f89cd3202c211a84b6ff099e44627bc20a404b1

Comment (TehDmitry) [reply index.user.js]: I just watch distance between center BH and enemies when enemies is moved inside BH, not just affected

Comment (EoD) [reply index.user.js]: Did you check if this works on very small and very high resolutions? I would expect it to require different numbers.

Comment (TehDmitry) [reply index.user.js]: You are right. Numbers will be different. 
I try to find solution.

**Comment by TehDmitry** [comment]
fixed. I hope.

**Pull request closed without merging**